charts/woodpecker-agent/values.yaml

```diff
@@ -31,13 +31,25 @@ nameOverride: ""
 fullnameOverride: ""
 
 serviceAccount:
-  # Specifies whether a service account should be created
+  # Specifies whether a service account should be created (also see RBAC subsection)
   create: true
   # Annotations to add to the service account
   annotations: {}
   # The name of the service account to use.
   # If not set and create is true, a name is generated using the fullname template
   name: ""
+  rbac:
+    # If your cluster has RBAC enabled and you're using the Kubernetes agent-
+    # backend you'll need this. (this is true for almost all production clusters)
+    create: false
```

Comment:
What about setting this to `true` directly?

Reply:
We could, but if I remember correctly deploying RBAC on a cluster without it enabled causes a failure, since the API-objects for role and roleBinding are non existant (could be wrong here haven't handled a RBAC less cluster in years)

Reply:
To be fair I've never worked with a cluster without, so I would be fine to risk that someone reports an issue / failure in case its not enabled.

Reply:
So... I've just proven my self wrong... pulled up a quick setup without RBAC enabled, and I could apply the role and roleBinding (without effect, but the API objects are there).

I've also just realized that I've missed a guard condition to not deploy the roleBinding and role when not creating a service account, although if we are changing the defaults, I would also go for always deploying the serviceAccount (this would give us a known set of allowed API-Requests).

Reply:
I've updated the guard condition and updated serviceAccount.rbac.create to true (also added a comment about disabling it and serviceAccount.create=true being neccessary for it to apply)